### PR TITLE
[NetAppFiles]Apply 2021-08-01 Changes

### PR DESCRIPTION
**Pull request by audunn**
## Description

Updates Azure NetApp Files to api-version 2021-01

API Specs
https://github.com/Azure/azure-rest-api-specs/pull/17157
https://github.com/Azure/azure-rest-api-specs/pull/16381

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1108
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/NetAppFiles/NetAppFiles/ActiveDirectory/NewNetAppFilesActiveDirectory.cs [hunk 1]
```diff
@@ -156,6 +155,18 @@ public class NewAzureRmNetAppFilesActiceDirectory : AzureNetAppFilesCmdletBase
...
+            Mandatory = false,
+            HelpMessage = "Domain Users to be added to the Built-in Administrators Active Directory group. A list of unique usernames without domain specifier.")]
+        [ValidateNotNullOrEmpty]
+        public string[] Administrators { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "If enabled, Traffic between the SMB server to Domain Controller (DC) will be encrypted.")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter EncryptDCConnections { get; set; }
```
Comment (wyunchi-ms) [inline src/NetAppFiles/NetAppFiles/ActiveDirectory/NewNetAppFilesActiveDirectory.cs:168]: As PowerShell suggest use singularity for parameters. You can find doc [here](https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/design-guidelines/parameter-best-practices.md#singularity). Please rename `Administrators` to `Administrator` and `EncryptDCConnections` to `EncryptDCConnection`.

File: src/NetAppFiles/NetAppFiles/ActiveDirectory/UpdateNetAppFilesActiveDirectory.cs [hunk 2]
```diff
@@ -168,6 +168,18 @@ public class UpdateAzureRmNetAppFilesActiveDirectory : AzureNetAppFilesCmdletBas
...
+            HelpMessage = "Domain Users to be added to the Built-in Administrators Active Directory group. A list of unique usernames without domain specifier.")]
+        [ValidateNotNullOrEmpty]
+        public string[] Administrators { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "If enabled, Traffic between the SMB server to Domain Controller (DC) will be encrypted.")]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter EncryptDCConnections { get; set; }
+
```
Comment (wyunchi-ms) [inline src/NetAppFiles/NetAppFiles/ActiveDirectory/UpdateNetAppFilesActiveDirectory.cs:182]: Same

File: src/NetAppFiles/NetAppFiles/Volume/NewNetAppFilesVolume.cs [hunk 3]
```diff
@@ -249,12 +249,30 @@ public class NewAzureRmNetAppFilesVolume : AzureNetAppFilesCmdletBase
...
+            Mandatory = false,
+            HelpMessage = "Volume spec name is the application specific designation or identifier for the particular volume in a volume group for e.g. data, log.")]
+        [ValidateNotNullOrEmpty]
+        public string VolumeSpecName { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Application specific placement rules for the particular volume.")]
+        [ValidateNotNullOrEmpty]
+        public IList<PSKeyValuePairs> PlacementRules { get; set; }
```
Comment (wyunchi-ms) [inline src/NetAppFiles/NetAppFiles/Volume/NewNetAppFilesVolume.cs:274]: As PowerShell suggest use singularity for parameters. You can find doc [here](https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/design-guidelines/parameter-best-practices.md#singularity). Please rename `PlacementRules` to `PlacementRule`.

File: src/NetAppFiles/NetAppFiles/ActiveDirectory/NewNetAppFilesActiveDirectory.cs (hunk 1 above)
Comment (audunn) [reply src/NetAppFiles/NetAppFiles/ActiveDirectory/NewNetAppFilesActiveDirectory.cs:168]: updated

**Comment by audunn** [comment]
@wyunchi-ms  Updated

File: src/NetAppFiles/NetAppFiles/help/New-AzNetAppFilesActiveDirectory.md [hunk 4]
```diff
@@ -77,6 +77,21 @@ Accept pipeline input: True (ByValue)
 Accept wildcard characters: False
 ```
 
+### -Administrators
```
Comment (wyunchi-ms) [inline src/NetAppFiles/NetAppFiles/help/New-AzNetAppFilesActiveDirectory.md:80]: Please update the document.

File: src/NetAppFiles/NetAppFiles/ActiveDirectory/NewNetAppFilesActiveDirectory.cs (hunk 1 above)
Comment (wyunchi-ms) [reply src/NetAppFiles/NetAppFiles/ActiveDirectory/NewNetAppFilesActiveDirectory.cs:168]: Do you?

**Comment by wyunchi-ms** [comment]
> @wyunchi-ms Updated

Do you push your commit? Please take a look at the status of CI pipeline.

File: src/NetAppFiles/NetAppFiles/ActiveDirectory/NewNetAppFilesActiveDirectory.cs (hunk 1 above)
Comment (audunn) [reply src/NetAppFiles/NetAppFiles/ActiveDirectory/NewNetAppFilesActiveDirectory.cs:168]: hmm, some git issue,  I will have a second look

Comment (audunn) [reply src/NetAppFiles/NetAppFiles/ActiveDirectory/NewNetAppFilesActiveDirectory.cs]: Is updating existing properties allowed ?

**Comment by audunn** [comment]
@wyunchi-ms git push solved and PR updated, thank you

**Review by wyunchi-ms — APPROVED** [review]

File: src/NetAppFiles/NetAppFiles/ActiveDirectory/NewNetAppFilesActiveDirectory.cs (hunk 1 above)
Comment (wyunchi-ms) [reply src/NetAppFiles/NetAppFiles/ActiveDirectory/NewNetAppFilesActiveDirectory.cs]: As your module is not GA yet. Is allowed.

**Pull request merged**